### ISSUE TITLE
the master version can`t support API below Level 11

### ISSUE THREAD

**Issue by zhanglt1986** [pakerfeldt/android-viewflow#95]
getMeasuredState() and MEASURED_STATE_TOO_SMALL are used in ViewFlow.java ,
 they are only support API level 11 and above.

**Comment by adouggy** [comment]
Yes, thanks. it works for me. default version from master is android 2.2 which make compile failed with hint getMeasuredState.

**Comment by guanguodong999** [comment]
Thanks , see this anwser .  issue is solved.

**Comment by yolapop** [comment]
Excuse me, what was the solution?

**Comment by guanguodong999** [comment]
oh,sorry,just saw. In your project,

bonne chance!

ÔÚ 2013-11-11 19:40:05£¬Yolapop notifications@github.com Ð´µÀ£º

Excuse me, what was the solution?

¡ª

**Comment by ghost** [comment]
What was the solution? I don't see anything between "in your project" and "bonne chance".

**Comment by guanguodong999** [comment]
That is a image between "in your project" and "bonne chance".
please check the enclosure.
eclipse -> viewflow-> project.properties
target = android-17

---

please check your project .perhaps your name of  project.properties file is  below API 11.
If you don't solve , you can email , and  attach the screenshot.
email : guanguodong999@163.com
ÔÚ 2013-11-15 07:31:40£¬"Charles Neveu" notifications@github.com Ð´µÀ£º

What was the solution? I don't see anything between "in your project" and "bonne chance".

¡ª

**Comment by Venni27** [comment]
Does it mean that the min sdk level is 11 now?

**Comment by siwangqishiq** [comment]
Does it mean that the min sdk level is 11 now?

**Comment by atest1** [comment]
branch:recyclebin is OK with API Level 8.

**Comment by ghost** [comment]
in ViewFlow.java function onMeasure()

replace
childState = child.getMeasuredState();

with
childState = child.MEASURED_HEIGHT_STATE_SHIFT >> child.MEASURED_STATE_MASK;

**Comment by Gurvinder1406** [comment]
Thanks a ton @PhilippEmmermacher 
Can you please briefly explain what exactly was the issue ?

**Comment by ghost** [comment]
@Gurvinder1406

The function getMeasuredState() was introduced in API 11.
This means it cannot be used in API's below Level 11.

But if you look at its documentation (http://developer.android.com/reference/android/view/View.html#getMeasuredState%28%29) you will easily notice what the function does:

"Return only the state bits of getMeasuredWidthAndState() and getMeasuredHeightAndState(), combined into one integer. The width component is in the regular bits MEASURED_STATE_MASK and the height component is at the shifted bits MEASURED_HEIGHT_STATE_SHIFT>>MEASURED_STATE_MASK. "

**Comment by ghost** [comment]
thanks a lot @PhilippEmmermacher that worked perfectly.
